libs/wallet/src/web3-react/hooks/useIsSmartContractWallet.ts

```diff
@@ -45,11 +45,8 @@ export function useIsSmartContractWallet(): boolean {
   useEffect(() => {
     if (!account) {
       setIsSmartContractWallet(false)
-      return
-    }
-
-    if (isAmbireWallet || isArgentWallet || isSmartContract) {
-      setIsSmartContractWallet(true)
+    } else {
```

Comment:
Is the code equivalent?

Reply:
Pretty much.
I just realized that we don't update the state in all variants of this useEffect.
Here I'm making sure that, if it's not a SC wallet, always set it to false.